scripts/TRN_TRN.R

```diff
@@ -0,0 +1,264 @@
+  ) |>
+  mutate(is_match_results_sponsor_protocol_id = TRUE) |>
+  select(trn1 = id, trn2 = results_sponsor_linked_trn, is_match_results_sponsor_protocol_id) |>
+  distinct(trn1, trn2, .keep_all = TRUE)
+
+# Integrate new results matches
+trn_trn_results_tidy <- trn_trn_protocols_tidy |>
+  mutate(is_match_results_sponsor_protocol_id = NA) |>
+  rows_upsert(trn_trn_results, by = c("trn1", "trn2")) |>
+  rowwise() |>
+  mutate(registry1 = case_when(is.na(registry1) ~ which_registry(trn1),
+                               .default = registry1),
+         registry2 = case_when(is.na(registry2) ~ which_registry(trn2),
+                               .default = registry2)
+  ) |>
+  ungroup()
+
+
+####################################################################################################################
+# Now add matches made from the title matching algorithm. Will follow similar logic to steps above
+
+# Modify structure of title match table to make it easier to integrate with larger match table
+trn_trn_titles <- title_matches |>
+  filter(id != euctr_id) |>  # filter out self-references
+  rename(trn1 = id, trn2 = euctr_id) |>
+  rowwise() |>
+  mutate(is_title_matched = TRUE,
+         trn1inreg2 = trn1 %in% trn_registries$trn2[trn_registries$trn1 == trn2],
+         # this next test is redundant because of circularity? default to TRUE?
+         trn2inreg1 = TRUE
+  ) |>
+  select(trn1, trn2, trn1inreg2, trn2inreg1, is_title_matched) |>
+  ungroup()
+
+# Integrate title matched TRNs
+trn_trn_titles_tidy <- trn_trn_results_tidy |>
+  mutate(is_title_matched = NA) |>
+  rows_upsert(trn_trn_titles, by = c("trn1", "trn2")) |>
+  rowwise() |> # added this rowwise()
+  mutate(registry1 = case_when(is.na(registry1) ~ which_registry(trn1),
+                               .default = registry1),
+         registry2 = case_when(is.na(registry2) ~ which_registry(trn2),
+                               .default = registry2)
+  )
+
+####################################################################################################################
+## Finally add matches from TRNs found in publications
+
+# Unnest TRNs in pubs table
+trn_trn_pubs_long <- publications |>
+  rename(trn1 = primary_IV_id) |>
+  mutate(trn_si_long = strsplit(trns_si, ";"),
+         trn_abs_long = strsplit(trns_abs, ";"),
+         trn_ft_long = strsplit(trns_ft, ";")) |> # Split trns_si by semicolon to get individual trial IDs
+  unnest_longer(trn_si_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_si_long != trn1 | is.na(trn_si_long)) |> # filter out self-references
+  unnest_longer(trn_abs_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_abs_long != trn1 | is.na(trn_abs_long)) |> # filter out self-references
+  unnest_longer(trn_ft_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_ft_long != trn1 | is.na(trn_ft_long)) |> # filter out self-references
+  select(trn1, contains("long"))
```

Comment:
You can make this more succinct. note that the variable name now should be changed for consistency:

```r
trn_trn_pubs_long <- publications |>
  rename(trn1 = primary_IV_id) |>
  separate_longer_delim(trns_si, delim =";") |> 
  separate_longer_delim(trns_abs, delim =";") |>
  separate_longer_delim(trns_ft, delim =";")
```